lib/crawler.js

```diff
@@ -958,6 +963,31 @@ Crawler.prototype.fetchQueueItem = function(queueItem) {
 };
 
 
+/*
+    Decode string buffer based on a complete Content-Type header. Will also look
+    for an embedded <meta> tag with a charset definition, but the Content-Type
+    header is prioritized, see:
+    https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#attr-charset
+
+    Examples
+
+        crawler.decodeBuffer(responseBuffer, "text/html; charset=tis-620");
+
+    Returns the decoded buffer.
+
+*/
+Crawler.prototype.decodeBuffer = function (buffer, contentTypeHeader) {
+    contentTypeHeader = contentTypeHeader || "";
+
+    var embeddedEncoding = /<meta.*charset=["']{0,1}([^"'>]*)["']{0,1}\s*\/{0,1}>/i.exec(buffer.toString(undefined, 0, 512)) || [],
+        encoding = contentTypeHeader.split("charset=")[1] || embeddedEncoding[1] || contentTypeHeader;
+
+    encoding = iconv.encodingExists(encoding) ? encoding : "utf-8";
```

Comment:
It seems that if `content-type =“ image / * ”` or any other response that is not a text file will be decoded into `utf-8`, in fact decode pictures, Word, Excel, PDF, ZIP and other files into` utf- 8` will cause the file to be damaged and cannot be opened.